addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/handler/LoxoneMiniserverHandler.java

```diff
@@ -0,0 +1,581 @@
+    }
+
+    /**
+     * Creates a new list of {@link Channel} for a single Loxone control object. Registers channel type within the
+     * factory, which is the channel type provider, or uses one of pre-registered type.
+     * Most of controls create only one channel, but some of them will create more channels to facilitate different
+     * types of states they support.
+     *
+     * @param control
+     *            control object to create a channel for
+     * @return
+     *         created list of {@link Channel} object
+     */
+    private List<Channel> createChannelsForControl(LxControl control) {
+
+        logger.trace("Creating channels for control: {}, {}", control.getClass().getSimpleName(), control.getUuid());
+
+        String label;
+        String controlUuid = control.getUuid().toString();
+        ChannelUID id = getChannelIdForControl(control, 0);
+
+        List<Channel> channels = new ArrayList<>();
+
+        LxCategory category = control.getCategory();
+
+        LxContainer room = control.getRoom();
+        String roomName = null;
+        if (room != null) {
+            roomName = room.getName();
+        }
+
+        String controlName = control.getName();
+        if (controlName == null) {
+            // Each control on a Miniserver must have a name defined, but in case this is a subject
+            // of some malicious data attack, we'll prevent null pointer exception
+            controlName = "Undefined name";
+        }
+
+        if (roomName != null) {
+            label = roomName + " / " + controlName;
+        } else {
+            label = controlName;
+        }
+
+        Set<String> tags = Collections.singleton("");
+
+        if (control instanceof LxControlPushbutton || control instanceof LxControlSwitch) {
+            if (category != null && category.getType() == LxCategory.CategoryType.LIGHTS) {
+                tags = Collections.singleton("Lighting");
+            }
+            addChannel(channels, "Switch", switchTypeId, id, label, "Switch", tags);
+        } else if (control instanceof LxControlJalousie) {
+            addChannel(channels, "Rollershutter", rollerTypeId, id, label, "Rollershutter", tags);
+        } else if (control instanceof LxControlInfoOnlyDigital) {
+            addChannel(channels, "Switch", roSwitchTypeId, id, label, "Digital virtual state", tags);
+        } else if (control instanceof LxControlInfoOnlyAnalog) {
+            LxControlInfoOnlyAnalog info = (LxControlInfoOnlyAnalog) control;
+            ChannelTypeUID typeId = addNewChannelType(control.getTypeName(), "Number", label, "Analog virtual state",
+                    info.getFormatString(), true, null, 0, controlUuid);
+            addChannel(channels, "Number", typeId, id, label, "Analog virtual state", tags);
+        } else if (control instanceof LxControlLightController) {
+            ChannelTypeUID typeId = addNewChannelType(control.getTypeName(), "Number", label, "Light controller", null,
```

Comment:
What is your suggestion on how to proceed here?
Note that my suggestion was to NOT dynamically create any channel types, but to simply define a static list (which already exists) and make use of those. It would make this binding much cleaner as you would not have to implement a ChannelTypeProvider at all.